common/src/app/common/geom/shapes/transforms.cljc

```diff
@@ -14,20 +14,23 @@
    [app.common.geom.shapes.path :as gpa]
    [app.common.geom.shapes.rect :as gpr]
    [app.common.math :as mth]
-   [app.common.pages.spec :as spec]
    [app.common.spec :as us]
    [app.common.text :as txt]))
 
+(def ^:dynamic *skip-adjust* false)
+
 ;; --- Relative Movement
 
-(defn- move-selrect [selrect {dx :x dy :y}]
-  (-> selrect
-      (d/update-when :x + dx)
-      (d/update-when :y + dy)
-      (d/update-when :x1 + dx)
-      (d/update-when :y1 + dy)
-      (d/update-when :x2 + dx)
-      (d/update-when :y2 + dy)))
+(defn- move-selrect [selrect pt]
+  (let [dx (.-x pt)
+        dy (.-y pt)]
+    (-> selrect
+        (update :x + dx)
+        (update :y + dy)
+        (update :x1 + dx)
+        (update :y1 + dy)
+        (update :x2 + dx)
+        (update :y2 + dy))))
```

Comment:
If the selrect is sensitive to performance, maybe we need to define a new type for selrect and this case, replace this update chain (that in this case will create at least 6 garbage objects). With a record we probably just create a single one with updated modifiers.

This is just a thinking, nothing definitive, we can talk about it and do it in other MR.

Reply:
I'll change the update for a new object creation so it has less objects